drivers/volume/portworx/portworx.go

```diff
@@ -820,6 +914,84 @@ func (p *portworx) VolumeDelete(pv *v1.PersistentVolume) error {
 	return p.volDriver.Delete(pv.Spec.PortworxVolume.VolumeID)
 }
 
+func (p *portworx) PerformRecovery() error {
+	var allSnaps *crdv1.VolumeSnapshotList
+	err := wait.ExponentialBackoff(defaultBackOff, func() (bool, error) {
```

Comment:
No need to retry here. If it fails return error. The init will fail and restart stork, which will cause the retry

Reply:
done